src/frontend/src/storage/index.ts

```diff
@@ -177,14 +179,20 @@ export const setKnownPrincipal = async ({
 
     // Add the new principal and sort (most recently used is first)
     dedupedPrincipals.push(principalData);
-    dedupedPrincipals.sort((a, b) => b.lastUsedTimestamp - a.lastUsedTimestamp);
+    dedupedPrincipals.sort(
+      (a, b) => (b.lastUsedTimestamp ?? 0) - (a.lastUsedTimestamp ?? 0)
+    );
 
     // Only keep the more recent N principals
     const prunedPrincipals = dedupedPrincipals.slice(0, MAX_SAVED_PRINCIPALS);
 
     // Here we try to be as non-destructive as possible and we keep potentially unknown
     // fields
-    storage.anchors[ix] = { ...oldAnchor, knownPrincipals: prunedPrincipals };
+    storage.anchors[ix] = {
+      ...oldAnchor,
+      knownPrincipals: prunedPrincipals,
+      lastUsedTimestamp: nowMillis(),
```

Comment:
This was missing. Setting a known principal didn't change the `lastTimeUsedTimestamp` which seems like a bug.